### ISSUE TITLE
no-implicit-dependencies ignoring whitelist elements

### ISSUE THREAD

**Issue by nixonad** [palantir/tslint#4058]
### Bug Report

- __TSLint version__: 5.11.0
- __TypeScript version__: 2.9.2
- __Running TSLint via__:  VSCode

#### TypeScript code being linted

```
import { file1 } from "@src/folder/file1";
import * as _ from "lodash";
```

with `tslint.json` configuration:

```
{
  "extends": ["tslint:recommended"],
  "rules": {
     "no-implicit-dependencies": [ true, "dev", ["@src"] ]
  }
}
```

and `package.json` configuration:

```
"devDependencies": {
    "tslint": "^5.11.0",
    "typescript": "^2.9.2"
  },
  "dependencies": { }
```

and `tsconfig.json` configuration (extract):

```
"compilerOptions": {
    "baseUrl": "./src",
    "paths": {
      "@src/*": [
        "*"
      ]
    },
    "module": "esnext",
...
```

#### Actual behavior
Both import statements are flagged as failing the no-implicit-dependencies rule.

#### Expected behavior
The lodash import statement should be flagged as failing and the @src absolute path import should not.

**Comment by xiongmao86** [comment]
According to @giladgray , the `@scopeuser/package` is the way npm refer to scope package and you should probably avoid using `@src` and whitelist `@src/folder` instead or using other setting like `#src` or `$src` in path instead.

**Comment by ajcrites** [comment]
I would recommend simply using `src`. There should be no confusion with respect to this name ... there is an npm module for it, “simple redis cache,” but it’s unlikely that you’re using it and it’s an old package.

**Comment by xiongmao86** [comment]
Got it. Thanks for correction, @ajcrites.

**Comment by JoshuaKGoldberg** [comment]
Thanks for the discussion folks - closing this as it seems the discussion has hit a logical conclusion.
